src/main/java/meditracker/command/CommandName.java

```diff
@@ -0,0 +1,44 @@
+package meditracker.command;
+
+/**
+ * The CommandName enum represents the names of commands supported by the application.
+ */
+public enum CommandName {
+    EXIT("exit"),
+    ADD("add"),
+    MODIFY("modify"),
+    LIST("list"),
+    DELETE("delete"),
+    SEARCH("search"),
+    TAKE("take"),
+    UNTAKE("untake");
+
+    public final String value;
+
+    /**
+     * Constructs a CommandName enum with the specified string value.
+     *
+     * @param value The string value associated with the command name.
+     */
+    CommandName(String value) {
+        this.value = value;
+    }
+
+    /**
+     * Returns the CommandName enum based on the provided string label.
+     *
+     * @param label The string label representing a command name.
+     * @return The CommandName enum corresponding to the label, or null if not found.
+     */
+    // @@author Baeldung
+    // Reused from https://www.baeldung.com/java-enum-values
+    // with minor modifications
+    public static CommandName valueOfLabel(String label) {
+        for (CommandName e : values()) {
+            if (e.value.equals(label)) {
+                return e;
+            }
+        }
+        return null;
+    }
```

Comment:
Good job on using Enums to handle the different `Command` types. Just a suggestion, since the value of the enum name are the same, you could just use `Enum.valueOf` and convert the label to `label.toUpperCase()` so you don't have to store additional attribute and also iterate through in a `for` loop.

```suggestion
    public static CommandName valueOfLabel(String label) {
        try {
            return CommandName.valueOf(label.toUpperCase());
        } catch (IllegalArgumentException e) {
            return null;
        }
    }
```

Reply:
Thanks for the suggestion. I have implemented and tried prior to this commit and faced other issues. One of it is the error when user supplied an invalid input, the toUpperCase() will fail and it leads to an exception which is the super class of NumberFormatException. Hence, the error handling may not be as specific and leads to other complications.